.github/workflows/ci_build_test.yml

```diff
@@ -92,7 +92,7 @@ jobs:
       - name: Rust Toolchain
         uses: dtolnay/rust-toolchain@master
         with:
-          toolchain: nightly-2024-10-17
+          toolchain: nightly-2025-04-03
```

Comment:
Update the nightly toolchain to have access to the newly stabilized `feature(trait_upcasting)`